sphinx_doc/build_instructions/build.rst

```diff
@@ -0,0 +1,235 @@
+Build Instructions
+##################
+
+Overview
+********
+
+The instructions to build **Navigation2 Stack** and **Navigation2 Dependencies**.
+
+- Build Options
+  
+  - `Quickstart Using Initial Setup Script`_
+  - `Manually Build From Source`_
+  
+- `Build Troubleshooting Guide`_
+
+**Recommended Platform**
+
+* `Ubuntu 18.04 <http://releases.ubuntu.com/18.04/ubuntu-18.04.3-desktop-amd64.iso>`_
+
+**Build Options**
+
+There two different recommended methods for building Navigation2 stack from source.
+
+**Easy Build**
+
+* An automated method to build Navigation2 in a simple and easy way.
+* To build Navigation2 by using the build script, go to `Quickstart using initial setup script`_
+
+Manually Build From Source
+**************************
+
+Step 1- Build ROS 2
+===================
+
+* `ROS2 Build Instructions <https://index.ros.org/doc/ros2/Installation>`_
+* When building ROS 2 from source, if you want to get the latest version of the ROS 2 packages, make sure that the `ros2.repos` file is from the `master` branch.
+* When installing the ROS 2 dependencies using rosdep, make sure to set `--rosdistro` to the lastest ROS 2 distribution (e.g `eloquent`)
+
+Step 2- Build Navigation2 Dependencies
+======================================
+
+- If you built ROS 2 from source, source the setup.bash file in the ROS 2 build workspace.
+  
+    ``source ~/ros2_ws/install/setup.bash``
+
+- If you installed ROS 2 from the binary packages, source the setup.bash file in the ROS 2 installation directory.
+    
+    ``source /opt/ros/<ros2-distro>/setup.bash``
+
+**Build and install Navigation2 dependencies:**
+
+.. code:: bash
+
+  mkdir -p ~/nav2_depend_ws/src
+  cd ~/nav2_depend_ws
+  wget https://raw.githubusercontent.com/ros-planning/navigation2/master/tools/ros2_dependencies.repos
+  vcs import src < ros2_dependencies.repos
+  rosdep install -y -r -q --from-paths src --ignore-src --rosdistro eloquent
+  colcon build --symlink-install
+
+
+Step 3- Build Navigation2
+=========================
+
+**I want to build the latest version:**
+
+- To build the lastest Navigation2 stack, set ``--branch`` to ``master``.
+- When installing dependencies using rosdep, make sure to set ``--rosdistro`` to the lastest ROS 2 distribution (e.g ``eloquent``)
+
+**I want to build an older version:**
+
+- To build Navigation2 stack for a specific ROS 2 distribution such as Cristal or Dashing, set ``--branch`` to the selected branch name. (e.g- ``dashing-devel``) 
+- Also you need to change ``--rosdistro`` to the selected ROS 2 distribution name. (e.g ``dashing``)
+
+.. code:: bash
+
+  source ~/nav2_depend_ws/install/setup.bash
+  mkdir -p ~/navigation2_ws/src
+  cd ~/navigation2_ws/src
+  git clone https://github.com/ros-planning/navigation2.git --branch master
+  cd ~/navigation2_ws
+  rosdep install -y -r -q --from-paths src --ignore-src --rosdistro eloquent
```

Comment:
Do we want to use `<ros2-distro>`? Maintain consistency with the [the turtlebot install instructions](https://github.com/ros-planning/navigation2/pull/1393/files#diff-9922005eb9744b16e9cfc99d325e62a1R94).

Reply:
In my opinion, 

``rosdep install -y -r -q --from-paths src --ignore-src --rosdistro eloquent``

is more copy paste friendly than 

``rosdep install -y -r -q --from-paths src --ignore-src --rosdistro <ros2-distro>``

Reply:
Or don't hard code at all `$ROS_DISTRO`

Reply:
From a user perspective, I'd get confused why it was `eloquent` in certain places and `<ros2-distro>` in few others. I personally prefer `<ros-distro>` unless we're tied to `eloquent`. Either way I think we should maintain the same convention throughout.

Reply:
I dont disagree with you @srsidd. I think that's something we can be better about. I think separately of this PR it would be helpful to glance over all the docs and make the changes to be consistent. Would love to see a PR for it!

Reply:
I think using the distro name in the build instructions is fine and makes the build easier for the user. For example, Steve's suggestion ```export ROS2_DISTRO=eloquent``` is still easy to maintain and use.

It should be <ros2-distro> in all the tutorials since it requires no maintenance.

Reply:
Its also not unreasonable for us to find-replace-all Dashing->Eloquent, Eloquent->Foxy, etc when we make a new branch. Its just on someone to actually... remember... to do it